.travis.yml

```diff
@@ -7,61 +7,76 @@ language: android
 
 before_install:
-- sed -i 's/git@github.com:/https:\/\/github.com\//' .gitmodules
-- git submodule update --init --recursive
-- chmod +x gradlew
-#- echo no | android create avd --force -n emulator-21-armv7a -t android-21 -b armeabi-v7a
-#- emulator -avd emulator-21-armv7a -no-skin -no-audio -no-window &
-#- android-wait-for-emulator
-- adb shell input keyevent 82 &
+  - sed -i 's/git@github.com:/https:\/\/github.com\//' .gitmodules
+  - git submodule update --init --recursive
+  - chmod +x gradlew
+  - wget https://dl.google.com/android/repository/android-ndk-r20-linux-x86_64.zip --quiet
+  - unzip -q android-ndk-r20-linux-x86_64.zip -d $HOME
+  - export ANDROID_NDK_HOME=$HOME/android-ndk-20
+  - export PATH=$PATH:$ANDROID_NDK_HOME
+  - mkdir "$ANDROID_HOME/licenses" || true
+  - echo -e "\n24333f8a63b6825ea9c5514f83c2829b004d1fee" > "$ANDROID_HOME/licenses/android-sdk-license"
+  - echo -e "\n84831b9409646a918e30573bab4c9c91346d8abd" > "$ANDROID_HOME/licenses/android-sdk-preview-license"
 install:
-- echo y | sdkmanager "ndk-bundle"
-- echo y | sdkmanager "cmake;3.6.4111459"
+
 before_script:
-- rm -fr $HOME/.gradle/caches/modules-2/files-2.1/com.bitmark.sdk
-- NDK_HOME=$ANDROID_NDK_HOME && SDK_HOME=$ANDROID_HOME && BINTRAY_USER=$BINTRAY_USER && BINTRAY_API_KEY=$BINTRAY_API_KEY && BINTRAY_GPG_PASS=$BINTRAY_GPG_PASS && BINTRAY_REPO=$BINTRAY_REPO
-- echo -n >local.properties && echo "ndk.dir=$NDK_HOME"$'\n'"sdk.dir=$SDK_HOME"$'\n'"bintray.user=$BINTRAY_USER"$'\n'"bintray.apikey=$BINTRAY_API_KEY"$'\n'"bintray.gpg.password=$BINTRAY_GPG_PASS"$'\n'"bintray.repo=$BINTRAY_REPO" >local.properties
-- chmod +x .travis/deploy.sh
+  - rm -fr $HOME/.gradle/caches/modules-2/files-2.1/com.bitmark.sdk
+  - NDK_HOME=$ANDROID_NDK_HOME && SDK_HOME=$ANDROID_HOME && BINTRAY_USER=$BINTRAY_USER && BINTRAY_API_KEY=$BINTRAY_API_KEY && BINTRAY_GPG_PASS=$BINTRAY_GPG_PASS && BINTRAY_REPO=$BINTRAY_REPO
+  - echo -n >local.properties && echo "ndk.dir=$NDK_HOME"$'\n'"sdk.dir=$SDK_HOME"$'\n'"bintray.user=$BINTRAY_USER"$'\n'"bintray.apikey=$BINTRAY_API_KEY"$'\n'"bintray.gpg.password=$BINTRAY_GPG_PASS"$'\n'"bintray.repo=$BINTRAY_REPO" >local.properties
+  - echo no | android create avd --force -n test -t android-24 --abi armeabi-v7a
+  - emulator -avd test -no-audio -no-window &
+  - android-wait-for-emulator
+  - adb shell input keyevent 82 &
 script:
-# Build & test
-- ./gradlew clean build
-#- ./gradlew clean test connectAndroidTest -p cryptography --info
-#- ./gradlew clean test connectAndroidTest -p android-sdk --info
-#- ./gradlew clean test connectAndroidTest -p cryptography --info
-#- ./gradlew clean build -p api-service --info
-#- ./gradlew clean build -p java-sdk --info
-#- ./gradlew clean test connectAndroidTest -p android-sdk --info
-
-# Deploy
-#- ./.travis/deploy.sh
+  - echo $PWD
+  - pushd cryptography
+  - pushd jni
+  - ./installswig.sh
+  - popd
+  - ./build-linux.sh || travis_terminate 1
+  - popd
+  - ./gradlew clean build -p api-service --info
```

Comment:
I'm not sure whether the `gradle build` command will run all tests? Just notice if you forgot.